### DISCUSSION TITLE
How to fetch data via useFetch or useAsyncData in nuxt3 from api in php?

### DISCUSSION THREAD

**Discussion by timoshechkin** [nuxt/nuxt#16465] [Questions]
Hello!
I have a PHP API on the server that generates the necessary data in JSON format. How can I get this data from PHP in NUXT3 through the useFetch or useAsyncData methods? What should a PHP script return when requested via useFetch or useAsyncData (data structure, headers, etc.)? Where it is possible to look at examples of such implementation?
Thanks in advance!

**Comment by TheAlexLichter** [comment]
Your API should just return the JSON you'd like to receive (+ likely indicating that it is JSON by setting the content type header). There is no special treatment necessary.

**Reply by timoshechkin** [reply]
Thank you. There are apparently some access problems on the local resource where the script is located. The response is returned: { "data": null, "pending": false, "error": "FetchError: Failed to fetch ()" }. Error description: "FetchError: 401 Unauthorized (http://10.250.150.90/test.php)". Tried hosting the script on an external site, everything worked great and got the JSON data.
Now it is not clear why a 401 error occurs. We need to somehow solve this problem.

**Reply by TheAlexLichter** [reply]
Could you kindly provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction) of the error you are experiencing? 🙏

**Reply by timoshechkin** [reply]
https://stackblitz.com/edit/github-f5ydlc?file=app.vue
I can't reproduce the 401 error, because the site with the PHP script is local.

**Reply by TheAlexLichter** [reply]
> [stackblitz.com/edit/github-f5ydlc?file=app.vue](https://stackblitz.com/edit/github-f5ydlc?file=app.vue) I can't reproduce the 401 error, because the site with the PHP script is local.

Could you host it somewhere then? That's the crucial part 🙈

**Reply by timoshechkin** [reply]
Hosted the PHP script on an external site:
http://timoshc3.beget.tech/test.php

**Reply by timoshechkin** [reply]
But when placing a PHP script on an external site, it works for me. I am getting a JSON response. I have a problem when hosting a PHP script on a local site.

**Reply by TheAlexLichter** [reply]
Which means the issue is about your local setup.

**Reply by timoshechkin** [reply]
Yes, I understand that I need to deal with access to a local resource. Tell me, is it possible to put a PHP script in the NUXT project directories? And how, in this case, to register the path to it in the useFetch request?
